### DISCUSSION TITLE
Following containerized way for settings

### DISCUSSION THREAD

**Discussion by iamrahul127** [DependencyTrack/dependency-track#1454] [Ideas]
I observed that many administrative setting are not exposed as environment variables or config file. Few examples setting "Internal Components", enabling "GitHub Advisories" and son on. My reflection is it would have been much nicer to configure them in values.yaml file than deploying and then configuring them using UI. Thoughts?

**Comment by stevespringett** [comment]
I don't think having credentials in a yaml file would be advisable. Docker secrets, etc, are possible and I think it would be good to expose some of the DT configuration as environment variables. However, this will become more maintenance over time as DT is broken down into more and more domain-specific services.

**Reply by iamrahul127** [reply]
In our case we deploy using azure pipelines and those secrets are passed through secret pipeline variables. So though we have values in values.yaml we use only them for testing and final deployment always go through azure pipeline. 
Challenge with current approach is we need to update these configuration after application is hosted though I understand it's one time activity if application is deployed in PROD environment.
